wolfssl/wolfcrypt/ecc.h

```diff
@@ -294,19 +295,16 @@ typedef struct ecc_set_type {
 
 /* determine max bits required for ECC math */
 #ifndef FP_MAX_BITS_ECC
-    /* check alignment */
-    #if ((MAX_ECC_BITS * 2) % DIGIT_BIT) == 0
-        /* max bits is double */
-        #define FP_MAX_BITS_ECC     (MAX_ECC_BITS * 2)
-    #else
-        /* max bits is doubled, plus one digit of fudge */
-        #define FP_MAX_BITS_ECC     ((MAX_ECC_BITS * 2) + DIGIT_BIT)
-    #endif
-#else
-    /* verify alignment */
-    #if FP_MAX_BITS_ECC % CHAR_BIT
-       #error FP_MAX_BITS_ECC must be a multiple of CHAR_BIT
-    #endif
+    /* ECC max bits must be multiple of 8 and include extra digit for math */
+    /* ROUND8(MAX_ECC_BITS) + ROUND8(DIGIT_BIT) */
+    #define FP_MAX_BITS_ECC  \
+        ((((MAX_ECC_BITS + (CHAR_BIT-1)) / CHAR_BIT) * CHAR_BIT) + \
```

Comment:
Code was 2 * MAX_ECC_BITS and is now just MAX_ECC_BITS + DIGIT_BITS.
Multiplication will required two times the size of the curve.
Is this correct?

Reply:
You are correct. The code and comment was something I added a long time ago and I believe it is wrong. After looking at the code and testing with real world use-cases it seems the required size is just ECC key size + one digit and some rounding (at least for fast math). It would be excellent to get your opinion on this.

Reply:
I did a test on Linux.
If FP_MAX_BITS_ECC is less than:
    (2 * ((MAX_ECC_BITS + DIGIT_BITS - 1) / DIGIT_BITS) * DIGIT_BITS)
then I get a failure.
(Actually I can subtract a DIGIT_BIT and it still works but I wouldn't do that.)

Reply:
@SparkiDev , thanks for confirming. I will revert and update this PR and the calcs.